tiles/src/main/java/com/protomaps/basemap/feature/Matcher.java

```diff
@@ -0,0 +1,255 @@
+package com.protomaps.basemap.feature;
+
+import com.onthegomap.planetiler.expression.Expression;
+import com.onthegomap.planetiler.expression.MultiExpression;
+import com.onthegomap.planetiler.reader.SourceFeature;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A utility class for matching source feature properties to values.
+ * 
+ * <p>
+ * Use the {@link #rule} function to create entries for a Planetiler {@link MultiExpression}. A rule consists of
+ * multiple contitions that get joined by a logical AND, and key-value pairs that should be used if all conditions of
+ * the rule are true. The key-value pairs of rules that get added later override the key-value pairs of rules that were
```

Comment:
Is this intentionally last-wins, or would it be cleaner to make earlier rules more important?